### PR TITLE
BCDA-4209 - Adding termination details to the ACO model

### PR DESCRIPTION
**Pull request by mtrang1263**
### Fixes [BCDA-4209](https://jira.cms.gov/browse/BCDA-4209)

To enable post termination access, we need to encode additional information about the termination, including the termination date. This PR adds a new field into our ACOs model to capture termination information.

With this new field, we will be deprecating the Blacklisted field in a follow up PR.

### Change Details
1. Added new termination type that captures the termination details listed in our design.
2. Introduced a termination type in the Postgres package that allows us to easily read/write this field to the ACOs table.
3. Cleaned up the postgresttest.go file to leverage the real query methods when possible.
4. Added schema migration to add this column to the ACOs table.

### Security Implications
- [ ] new software dependencies
- [ ] security controls or supporting software altered
- [x] new data stored or transmitted
We are going to store additional information around ACO termination. No affect on PHI/PIII.

- [ ] security checklist is completed for this change
- [ ] requires more information or team discussion to evaluate security implications
- [x] no PHI/PII is affected by this change

### Acceptance Validation
1. CI Passes.
2. Run up migration on dev environment and ensure that the tests pass as expected.

### Feedback Requested

Let me know what your thoughts are on this file: `bcda/models/postgres/termination.go`. It implements the Scanner and Valuer interface allow us to easily read/write this field.

### CODE REVIEW

**Review by SMLuthi — APPROVED** [review]
This LGTM! I especially like the repository clean up, much cleaner.

File: bcda/models/termination.go [hunk 1]
```diff
@@ -0,0 +1,106 @@
+package models
+
+import (
+	"fmt"
+	"time"
+)
+
+type Termination struct {
+	Date                time.Time
```
[the remark this answers is not in the document]
Comment (mtrang1263) [reply bcda/models/termination.go:9]: Great idea! Will update the naming convention here. Date == TerminationDate and will add CutoffDate which represents the date where they caller will no longer have access.

**Review by katyasoup — APPROVED** [review]
LGTM!

**Comment by CosmicSyntax** [comment]
LGTM!

**Review by SMLuthi — APPROVED** [review]

**Review by CosmicSyntax — APPROVED** [review]
LGTM

**Review by CosmicSyntax — APPROVED** [review]
Looks great!

**Pull request merged**